189_leet_rotatearray.py

```diff
@@ -0,0 +1,26 @@
+from typing import List
+
+
+class Solution:
+    def rotate(self, nums: List[int], k: int) -> None:
+        """
+        Do not return anything, modify nums in-place instead.
+        """
+        # reverse the input array
+        nums.reverse()
+        k = k % len(nums)
+        # reverse the first k elements
+        start = 0
+        end = k - 1
+        while start < end:
+            nums[start], nums[end] = nums[end], nums[start]
+            start += 1
+            end -= 1
```

Comment:
nit: This is repeated 2 time, you can do a small decomposition and move this to a separate function, for example:

```
def reverse(nums: List[int], start: int, end: int) -> None:
        while start < end:
            nums[start], nums[end] = nums[end], nums[start]
            start += 1
            end -= 1
```
and then just do `reverse(nums, start, end)`

It is a bit of overengineering for this particular usecase, but it is a good way to look a bit better at the coding interview